robot/basestation/static/js/elements/feed/console.js

```diff
@@ -13,63 +13,67 @@ $(document).ready(() => {
           pingOdroid()
           break
       }
-      lastCmdSent = new Date().getTime()
+      setTimeSinceCommand()
     }
   }
 
   function pingOdroid (timeoutVal = REQUEST_TIMEOUT) {
-    appendToConsole('pinging odroid')
-    $.ajax('/ping_rover', {
-      success: function (data) {
-        appendToConsole(data.ping_msg)
-        if (!data.ros_msg.includes('Response')) {
-          appendToConsole('No response from ROS ping_acknowledgment service')
-        } else {
-          appendToConsole(data.ros_msg)
-        }
-      },
-      error: function (jqXHR, textStatus, errorThrown) {
-        console.log(errorThrown)
-        if (errorThrown == 'timeout') {
-          msg =
-            'Odroid ping timeout after ' +
-            timeoutVal / 1000 +
-            ' seconds. ' +
-            'Check if the websockets server is running. If not, there\'s either a network issue ' +
-            'or the Odroid and possibly the whole rover has shut down unexpectedly.'
-          appendToConsole(msg)
-        } else {
-          console.log('Error of type ' + errorThrown + 'occured')
-        }
-      },
-      timeout: timeoutVal
-    })
-    lastCmdSent = new Date().getTime()
+    if (canSendCommand(PING_THROTTLE_TIME)) {
+      appendToConsole('pinging odroid')
+      $.ajax('/ping_rover', {
+        success: function (data) {
+          appendToConsole(data.ping_msg)
+          if (!data.ros_msg.includes('Response')) {
+            appendToConsole('No response from ROS ping_acknowledgment service')
+          } else {
+            appendToConsole(data.ros_msg)
+          }
+        },
+        error: function (jqXHR, textStatus, errorThrown) {
+          console.log(errorThrown)
+          if (errorThrown == 'timeout') {
+            msg =
+              'Odroid ping timeout after ' +
+              timeoutVal / 1000 +
+              ' seconds. ' +
+              'Check if the websockets server is running. If not, there\'s either a network issue ' +
+              'or the Odroid and possibly the whole rover has shut down unexpectedly.'
+            appendToConsole(msg)
+          } else {
+            console.log('Error of type ' + errorThrown + 'occured')
+          }
+        },
+        timeout: timeoutVal
+      })
+    }
+    setTimeSinceCommand()
```

Comment:
This seems like duplicate code, same exact thing as in app.js. Can we remove one of these instances? :eyes: